### PR TITLE
Add select component

### PR DESCRIPTION
**Pull request by steveoh**
And a few other fixes mixed in

### CODE REVIEW

**Review by stdavis — COMMENTED** [review]
The new icons look much better on my non-hd screens. ⭐

Question: How do I make the select smaller than the default size? I would like to fit it inline and it's a bit too large.

Otherwise, the Select seems to work great!

**Comment by steveoh** [comment]
What size do you want it to be?

**Comment by stdavis** [comment]
I tried setting the width via the className prop and it didn't work. I'll give you an example next week. I wondered if it had to do with the fit-contents width value that I saw in there somewhere. But I'm not sure.

**Comment by steveoh** [comment]
There's a min width you'd want to override

**Comment by steveoh** [comment]
Let's remove the min width and see how that treats us.

**Comment by stdavis** [comment]
That was the trick!
image

What about a min-height to make it the same height as an input if there is an empty value?

**Comment by stdavis** [comment]
Adding `min-h-[34px]` seems to work for me but I'm not sure if that's what you want to do.

**Comment by stdavis** [comment]
Looking good!
image

**Review by stdavis — COMMENTED** [review]
I'm getting the "Component auth has not been registered yet" error when I attempted to use the firebase provider components from kitchen sink. Did you get this working with a separate app yet?

File: packages/utah-design-system/src/components/Select.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  variants: {
+    isDisabled: {
+      false:
+        'text-gray-800 group-invalid:border-rose-600 hover:bg-gray-100 pressed:bg-gray-200 dark:text-zinc-300 dark:hover:bg-zinc-600 dark:pressed:bg-zinc-500 forced-colors:group-invalid:border-[Mark]',
+      true: 'border-gray-200 text-gray-200 dark:border-zinc-700 dark:bg-zinc-900 dark:text-zinc-600',
+    },
+  },
+});
+
+export interface SelectProps<T extends object>
```
Comment (stdavis) [inline packages/utah-design-system/src/components/Select.tsx:30]: What does `T extends object` do? I may need some educating...

**Comment by stdavis** [comment]
I'm getting a strange error message when I attempt to use the built version:
image

I can't imagine why TS thinks that those props are required...

**Review by stdavis — APPROVED** [review]
I'm having all sorts of issues trying to npm link this package for testing in electrofishing. I say that we merge and deploy and fix anything that comes up.

**Pull request merged**